armory/utils/export.py

```diff
@@ -242,26 +242,26 @@ def get_coco_formatted_bounding_box_data(
         ground_truth_boxes_coco_format = []
         predicted_boxes_coco_format = []
 
-        image_id = y_i["image_id"][0]  # All boxes in y_i are for the same image
-
-        if y_i is not None:
-            bboxes_true = y_i["boxes"]
-            labels_true = y_i["labels"]
+        if y_i is None:
+            # We can't export anything without y_i, because
+            # even exporting y_i_pred requires y_i['image_id'].
+            log.warning("Cannot export bounding box annotations due to missing image_id")
+            return ground_truth_boxes_coco_format, predicted_boxes_coco_format
```

Comment:
I think we could make `y_i` a required arg since it's necessary for this method to work (as opposed to having a warning and the function returning early)